Mathlib/Analysis/NormedSpace/PiTensorProduct/InjectiveSeminorm.lean

```diff
@@ -0,0 +1,467 @@
+    simp only [Seminorm.comp_apply, coe_normSeminorm, ge_iff_le]
+    rw [ContinuousLinearMap.opNorm_le_iff (Finset.prod_nonneg (fun _ _ ↦ norm_nonneg _))]
+    intro f
+    simp only [toDualContinuousMultilinearMap_apply_toFun, lift.tprod,
+        ContinuousMultilinearMap.coe_coe]
+    rw [mul_comm]
+    exact ContinuousMultilinearMap.le_opNorm f m
+
+noncomputable instance : SeminormedAddCommGroup (⨂[𝕜] i, E i) :=
+  AddGroupSeminorm.toSeminormedAddCommGroup injectiveSeminorm.toAddGroupSeminorm
+
+noncomputable instance : NormedSpace 𝕜 (⨂[𝕜] i, E i) where
+  norm_smul_le a x := by
+    change injectiveSeminorm.toFun (a • x) ≤ _
+    rw [injectiveSeminorm.smul']
+    rfl
+
+variable (𝕜 E F)
+
+/-- The linear equivalence between `ContinuousMultilinearMap 𝕜 E F` and `(⨂[𝕜] i, Eᵢ) →L[𝕜] F`
+induced by `PiTensorProduct.lift`, for every normed space `F`.
+-/
+@[simps]
+noncomputable def liftEquiv : ContinuousMultilinearMap 𝕜 E F ≃ₗ[𝕜] (⨂[𝕜] i, E i) →L[𝕜] F where
+  toFun f := LinearMap.mkContinuous (lift f.toMultilinearMap) ‖f‖
+    (fun x ↦ injectiveSeminorm_bound f x)
+  map_add' f g := by ext _; simp only [ContinuousMultilinearMap.toMultilinearMap_add, map_add,
+    LinearMap.mkContinuous_apply, LinearMap.add_apply, ContinuousLinearMap.add_apply]
+  map_smul' a f := by ext _; simp only [ContinuousMultilinearMap.toMultilinearMap_smul, map_smul,
+    LinearMap.mkContinuous_apply, LinearMap.smul_apply, RingHom.id_apply,
+    ContinuousLinearMap.coe_smul', Pi.smul_apply]
+  invFun l := MultilinearMap.mkContinuous (lift.symm l.toLinearMap) ‖l‖ (fun x ↦ by
+    simp only [lift_symm, LinearMap.compMultilinearMap_apply, ContinuousLinearMap.coe_coe]
+    refine le_trans (ContinuousLinearMap.le_opNorm _ _) (mul_le_mul_of_nonneg_left ?_
+      (norm_nonneg l))
+    exact injectiveSeminorm_tprod_le x)
+  left_inv f := by ext x; simp only [LinearMap.mkContinuous_coe, LinearEquiv.symm_apply_apply,
+      MultilinearMap.coe_mkContinuous, ContinuousMultilinearMap.coe_coe]
+  right_inv l := by
+    rw [← ContinuousLinearMap.coe_inj]
+    apply PiTensorProduct.ext; ext m
+    simp only [lift_symm, LinearMap.mkContinuous_coe, LinearMap.compMultilinearMap_apply,
+      lift.tprod, ContinuousMultilinearMap.coe_coe, MultilinearMap.coe_mkContinuous,
+      ContinuousLinearMap.coe_coe]
+
+/-- The equivalence `PiTensorProduct.liftEquiv` as an isometric linear equivalence.
+-/
+@[simps!]
+noncomputable def liftIsometry  : ContinuousMultilinearMap 𝕜 E F ≃ₗᵢ[𝕜] (⨂[𝕜] i, E i) →L[𝕜] F :=
+  {liftEquiv 𝕜 E F with
+   norm_map' := by
+     intro f
+     refine le_antisymm ?_ ?_
+     · simp only [liftEquiv_apply]
+       exact LinearMap.mkContinuous_norm_le _ (norm_nonneg f) _
+     · conv_lhs => rw [← (liftEquiv 𝕜 E F).left_inv f]
+       simp only [LinearEquiv.invFun_eq_symm, liftEquiv_symm_apply]
+       exact MultilinearMap.mkContinuous_norm_le _ (norm_nonneg _) _}
+
+variable {E F}
+
+/-- The canonical continuous multilinear map from `E` to `⨂[𝕜] i, Eᵢ`.
```

Comment:
Maybe write
"from `E = E₁ × ... × Eₙ` "
(here and elsewhere if it can help, maybe?)

Reply:
Ok.